DropException.cs

```diff
@@ -0,0 +1,16 @@
+using System;
+using System.Runtime.Serialization;
+
+namespace DropDownloadCore
+{
+    public class DropException : Exception
+    {
+        public DropException() { }
+
+        public DropException(string message) : base(message) { }
+
+        public DropException(string message, Exception innerException) : base(message, innerException) { }
+
+        protected DropException(SerializationInfo info, StreamingContext context) : base(info, context) { }
```

Comment:
What's the extra overrides that aren't used? Expecially the protected one.

Reply:
Boilerplate Exception snippet. The protected one is used when serializing Exception objects either across a wire or across appdomains (not relevant here but not harmful)

Reply:
I think I have mixed feelings about app specific exceptions especially if there is this much boilerplate but not a biggy.

Reply:
I used this to replace generic Exception objects being thrown -- I'm fine with piggybacking on others. I'll take a look at usage and maybe back this out in favor of other stuff.

Reply:
threw them out and a lot of validations that weren't actually necessary.